src/std/os/fdio.ss

```diff
@@ -43,6 +43,21 @@
     (do-retry-nonblock (_close raw)
       (close raw))))
 
+(def (flush raw)
+  (if (fd? raw)
+    (check-os-error (_flush raw)
+      (flush raw))
+    (force-output raw 2)))
+
+(def (fdseek raw position from)
+  (let ((fd (if (fd? raw) (fd-e raw) raw))
+        (whence (match from
```

Comment:
you can also use case here, but please add an else clause that raises a useful error.